integrations/opentelemetry-wsgi/src/opentelemetry_wsgi/__init__.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The opentelemetry_wsgi integration provides a WSGI middleware that can be used
+on any WSGI framework (such as Django / Flask) to track requests timing through
+OpenTelemetry.
+"""
+
+import opentelemetry.trace as trace
+
+
+class OpenTelemetryMiddleware:
+    """The WSGI application middleware.
+
+    This class is used to create and annotate spans for requests to a WSGI
+    application.
+
+    :param wsgi: The WSGI application callable.
+    """
+
+    def __init__(self, wsgi):
+        self.wsgi = wsgi
+        self.start_response = None
+        self.span = None
+
+    def _add_request_attributes(self, environ):
+        self.span.add_attribute("http.method", environ["HTTP_METHOD"])
+        self.span.add_attribute("http.path", environ["PATH_INFO"])
+        self.span.add_attribute("http.route", environ["PATH_INFO"])
```

Comment:
Please refer to https://github.com/census-instrumentation/opencensus-python/pull/759.